composer.json

```diff
@@ -134,28 +134,6 @@
     },
     "minimum-stability": "stable",
     "prefer-stable": true,
-    "scripts": {
-        "post-install-cmd": [
-            "php bin/console extensions:configure --with-config",
-            "@auto-scripts",
-            "php bin/console bolt:info"
-        ],
-        "pre-package-uninstall": [
-            "php bin/console extensions:configure --remove-services"
-        ],
-        "post-update-cmd": [
-            "php bin/console extensions:configure",
-            "@auto-scripts",
-            "php bin/console bolt:info"
-        ],
-        "auto-scripts": {
-            "cache:clear --no-warmup": "symfony-cmd",
-            "assets:install --symlink --relative %PUBLIC_DIR%": "symfony-cmd"
-        },
```

Comment:
Should be part of website build script or Dockerfile, not core code `composer.json`

Reply:
I'm not sure about this specific change. As a developer, I use `bolt/core` directly when working _on_ Bolt. Did you remove it because it's problematic, or just as part of cleaning up?

Reply:
Understood.

This part it problematic, as it requires whole project to be runnable, before even running codding standard or PHPStan or PHP linter in CI.

So if code is broken, the composer install will fail here and no CI tool will help us.

Reply:
Is that really a problem, though? If the code is broken, CI _should_ fail, and we'd need to fix that regardless.

Reply:
It is, because it fails on `composer install` and not on one of 8 jobs we have now. It's false positive actually. The point of CI is to have the minimal scope the error appears in. That's why the PHPStan should be decouple from project's container.

In a correct project setup, this shoud be part of build script, Docker, `npm run` etc. Same way assets build is.

Reply:
I see your point. Bluntly removing them isn't the answer though, because we use these in our daily workflow. Maybe we can have composer run with `--no-scripts` in CI?

Reply:
It seems like an workaround of current issue, which stack on incorrect use of composer scripts.
But if that's the least we can do right now, let's do it. We can handle it later better.

Reply:
I've pushed it now, let's see how CI handles it :)

Reply:
Hm, all but 2 jobs fails now :D

Reply:
Several are breaking with the following error 🤔 

```
 PHP Fatal error:  Uncaught Symfony\Component\Dotenv\Exception\PathException: Unable to read the "/home/runner/work/core/core/bin/../.env" environment file. in /home/runner/work/core/core/vendor/symfony/dotenv/Dotenv.php:565
```

Reply:
That's the problem I got at first place. The .env file should not be required to install dependencies. Only after

This the issue stacking I talked about. --no-scripts probably disables some process outside `composer.json` that some package rely on by convention.

Reply:
Right, it's breaking here: https://github.com/bolt/core/blob/master/bin/console#L18-L23

```php
if (!isset($_SERVER['APP_ENV'])) {
    if (!class_exists(Dotenv::class)) {
        throw new \RuntimeException('APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.');
    }
    (new Dotenv())->load(__DIR__.'/../.env');
}
```

`$_SERVER['APP_ENV']` isn't set, so it attempts to read `.env`, which doesn't exist.

In Travis the env settings are set from `.env.dist` by: 

```bash
export $(grep -v '^#' .env.dist | xargs -d '\n')
```

Should the github workflow do something similar?

Reply:
Why not have the file with name `.env` by default? It would remove this hidden dependency.

The convention is to have prod-based configs if needed.

- .env.prod
- .env.local
- .env.test

etc

See Symfony post about it: https://symfony.com/doc/4.1/configuration/dot-env-changes.html

Reply:
I commited it, just to try it out. I don't think it will be enough